internal/provider/resource_tfe_notification_configuration.go

```diff
@@ -94,333 +245,295 @@ func resourceTFENotificationConfiguration() *schema.Resource {
-		// When destination_type is 'email':
-		// 1. url and token cannot be set
-		err := validateSchemaAttributesForDestinationTypeEmail(d)
-		if err != nil {
-			return err
-		}
-	} else if destinationType == tfe.NotificationDestinationTypeGeneric {
-		// When destination_type is 'generic':
-		// 1. email_addresses and email_user_ids cannot be set
-		// 2. url must be set
-		err := validateSchemaAttributesForDestinationTypeGeneric(d)
-		if err != nil {
-			return err
-		}
-	} else if destinationType == tfe.NotificationDestinationTypeSlack {
-		// When destination_type is 'slack':
-		// 1. email_addresses, email_user_ids, and token cannot be set
-		// 2. url must be set
-		err := validateSchemaAttributesForDestinationTypeSlack(d)
-		if err != nil {
-			return err
-		}
-	} else if destinationType == tfe.NotificationDestinationTypeMicrosoftTeams {
-		// When destination_type is 'microsoft-teams':
-		// 1. email_addresses, email_user_ids, and token cannot be set
-		// 2. url must be set
-		err := validateSchemaAttributesForDestinationTypeMicrosoftTeams(d)
-		if err != nil {
-			return err
-		}
+// Create implements resource.Resource
+func (r *resourceTFENotificationConfiguration) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, config modelTFENotificationConfiguration
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
+	workspaceID := plan.WorkspaceID.ValueString()
+
 	// Create a new options struct
 	options := tfe.NotificationConfigurationCreateOptions{
-		DestinationType: tfe.NotificationDestination(destinationType),
-		Enabled:         tfe.Bool(enabled),
-		Name:            tfe.String(name),
-		Token:           tfe.String(token),
-		URL:             tfe.String(url),
+		DestinationType: tfe.NotificationDestination(tfe.NotificationDestinationType(plan.DestinationType.ValueString())),
+		Enabled:         plan.Enabled.ValueBoolPointer(),
+		Name:            plan.Name.ValueStringPointer(),
+		URL:             plan.URL.ValueStringPointer(),
+		SubscribableChoice: &tfe.NotificationConfigurationSubscribableChoice{
+			Workspace: &tfe.Workspace{ID: workspaceID},
+		},
+	}
+
+	// Set Value from `token_wo` if set, otherwise use the normal value
+	if !config.TokenWO.IsNull() {
+		options.Token = config.TokenWO.ValueStringPointer()
+	} else {
+		options.Token = plan.Token.ValueStringPointer()
 	}
 
 	// Add triggers set to the options struct
-	for _, trigger := range d.Get("triggers").(*schema.Set).List() {
-		options.Triggers = append(options.Triggers, tfe.NotificationTriggerType(trigger.(string)))
+	var triggers []types.String
+	if diags := plan.Triggers.ElementsAs(ctx, &triggers, true); diags != nil && diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
```

Comment:
Definitely no problem here, just another way I have seen this written, choose whichever you like:
```suggestion
	resp.Diagnostics.Append(plan.Triggers.ElementsAs(ctx, &triggers, true)...)
	if resp.Diagnostics.HasError() {
		return
	}
```